nntrainer/src/flatten_layer.cpp

```diff
@@ -59,7 +59,7 @@ Tensor FlattenLayer::backwarding(Tensor in, int iteration) {
 
 void FlattenLayer::setProperty(const PropertyType type,
                                const std::string &value) {
-  throw std::invalid_argument("[Flatten Layer] setProperty not supported");
+  throw std::out_of_range("[Flatten Layer] setProperty not supported");
```

Comment:
out_of_range? IMO, out of range is for accessing out of bounds.

Reply:
There are two case for `std::invalid_argument`
1. when property type is not valid 🌵 
2. when given value is not valid 🍰 

I needed to differentiate 🌵 and 🍰 .
While it'd be better off to derive a custom exception to represent the case. I just made 🌵 `std::out_of_range` to save time considering that accessing invalid property is kinda similar to accessing out of bounds.

flatten layer case is 🌵 since every property type is invalid.

```cpp
  /**
   * ...
   * @exception std::out_of_range     when property type is not valid for the
   * particular layer
   * @exception std::invalid_argument invalid argument
   */
  virtual void setProperty(const PropertyType type,
                           const std::string &value = "");
```

Reply:
As the handling of this case (`out_of_range`) is very simple (empty), I think it's not a big overhead to make a custom exception. It would make code more readable.

Reply:
Making it is not so hard though, If we are going to make a custom exception it should be managed globally because it need be caught by `nntrainer_exception_boundary`.  Considering this, do you have opinion to where to put custom exceptions? maybe `nntrainer_error.h`?

Reply:
nntrainer_error.h or util_func.h

Reply:
Alright, I'll do that in other PR to control size